objc/objc_runtime_darwin.go

```diff
@@ -215,6 +225,17 @@ func (i Ivar) Offset() uintptr {
 	return ret
 }
 
+// Protocol is a type that declares methods that can be implemented by any class.
+type Protocol uintptr
+
+// GetProtocol returns the protocol for the given name or nil if there is no protocol by that name.
+func GetProtocol(name string) *Protocol {
+	n := strings.CString(name)
+	p, _, _ := purego.SyscallN(objc_getProtocol, uintptr(unsafe.Pointer(n)))
+	runtime.KeepAlive(n)
+	return *(**Protocol)(unsafe.Pointer(&p))
```

Comment:
Isn't this:

```go
return (*Protocol)(unsafe.Pointer(p))
```

?

Reply:
go vet will complain that it is potentially unsafe conversion which is why I did the indirection

Reply:
I see. Syscall doesn't work well `go vet` actually (https://github.com/golang/go/issues/41205)

It is fine to not doing go-vet, but what do you think?

If we keep go-vet and use the indirection way, let's leave a comment for the intention.